persistence/provider/memory/eventstore.go

```diff
@@ -126,6 +132,8 @@ func (cs *eventStoreChangeSet) stageSave(
 
 	cs.items = append(cs.items, item)
 
+	result.EventItems = append(result.EventItems, item)
```

Comment:
I think you can handle this entirely in `Commit()`. You have access to the `eventStoreChangeSet` there, so you can just use `cs.items` as `EventItems` in the result. There are a few benefits to doing so:

- this code wont need to change when `SaveEvent()` loses it's offset return value (#168 
- avoid append()'ing
- don't need `transaction.result` field anymore

Reply:
Cool, i will do this in a follow-up commit.